lib/ruby_llm/chat.rb

```diff
@@ -26,16 +26,22 @@ def initialize(model: nil, provider: nil)
     end
 
     def ask(message = nil, with: {}, &block)
-      add_message role: :user, content: Content.new(message, with)
+      # Extract cache_control from the with hash if present
```

Comment:
I'm unsure if this is how we want to handle things. this essentailly adds global functionality for an anthropic specific concept.